rationale.md

```diff
@@ -383,42 +427,192 @@ async create_query_task(
+
+Until we've [resolved this issue](#binary-response-detection) with replacement HTTP request methods, the code generator will assist with binary responses in the following ways (the provided examples in Typescript).
+
+#### Response can be either binary and string
+
+A **mixed mode** response that can be either binary or string (e.g. text or json) has a note placed into the method header.
+
+```typescript
+/**
+  * POST /queries/run/{result_format} -> string}
+  *
+  * **Note**: Binary content may be returned by this method. Add `binaryMode` to the `options` parameter to correctly receive binary content
+  */
+async run_inline_query(
+  request: Partial<IRequestrun_inline_query>,
+  options?: Partial<ITransportSettings>,
+) {
+  return this.post<string, IError>(
+    encodeURI(`/queries/run/${request.result_format}`),
+    {
+      limit: request.limit,
+      apply_formatting: request.apply_formatting,
+      apply_vis: request.apply_vis,
+      cache: request.cache,
+      image_width: request.image_width,
+      image_height: request.image_height,
+      generate_drill_links: request.generate_drill_links,
+      force_production: request.force_production,
+      cache_only: request.cache_only,
+      path_prefix: request.path_prefix,
+      rebuild_pdts: request.rebuild_pdts,
+      server_table_calcs: request.server_table_calcs,
+    },
+    request.body,
+    options,
+  )
+}
+```
+
+#### Binary responses
+
+A method that only has binary responses will have the request configured for binary inside the generated SDK method. Images and PDFs are considered binary.
+
+```typescript
+/**
+  * GET /render_tasks/{render_task_id}/results -> string}
+  *
+  * **Note**: Binary content is returned by this method.
+  */
+async render_task_results(
+  /**
+    * @param {string} render_task_id Id of render task
+    */
+  render_task_id: string,
+  options?: Partial<ITransportSettings>,
+) {
+  options = { options, ...binaryMode }
+  return this.get<string, IError>(
+    encodeURI(`/render_tasks/${render_task_id}/results`),
+    null,
+    null,
+    options,
+  )
+}
+```
+
+#### String responses
+
+No changes are made to the standard generated method for response that are handled as string, like text or json
+
+```typescript
+/**
+  * GET /roles/{role_id} -> IRole}
+  */
+async role(
+  /**
+    * @param {number} role_id id of role
+    */
+  role_id: number,
+  options?: Partial<ITransportSettings>,
+) {
+  return this.get<IRole, IError>(
+    encodeURI(`/roles/${role_id}`),
+    null,
+    null,
+    options,
+  )
+}
+```
+
+### Future enhancements
+
+While the new SDKs are easier to use than the previous options, there are still some additional improvements we are working on.
+
+#### Streaming support
+
+Looker intends to provide streaming (http chunking) methods for all streamable API endpoints, in addition to the current non-streaming methods.
```

Comment:
If we're going to mention it, perhaps a little more explanation of what streaming is good for?

> Assigning an API response to a local variable is simple and easy, but if the response is extremely large (gigabytes of data), deserializing into client app memory will probably cause the client app to crash with an out-of-memory exception. Streaming large responses chunk by chunk can achieve constant memory use for arbitrarily large responses, at a cost of additional code complexity.

Reply:
added